### PR TITLE
Feat: Added filters on product page

### PR DESCRIPTION
**Pull request by abhijeetekad**
Added Filters on the Product page.
Filter by price range.
Filter by category.
Filter by rating
chrome-capture (11)
.

### CODE REVIEW

**Review by SukanyaSen530 — APPROVED** [review]

File: src/Components/ProductList/ProductList.jsx [hunk 1]
```diff
@@ -1,14 +1,43 @@
+import { useFilter } from "../../Context/FilterContext";
 import { useProduct } from "../../Context/ProductContext";
 import { ProductCard } from "../ProductCard/ProductCard";
 import "./ProductList.css";
 const ProductList = () => {
   const { productList } = useProduct();
+  const { state } = useFilter();
+  console.log(state);
```
Comment (prathmeshjagtap) [inline src/Components/ProductList/ProductList.jsx]: remove console.log

File: src/Components/ProductList/ProductList.jsx [hunk 2]
```diff
@@ -1,14 +1,43 @@
...
+  const filterByPrice = (products, sortByPrice) => {
+    if (sortByPrice === "lowToHigh") {
+      return products.sort((a, b) => a.discountedPrice - b.discountedPrice);
+    } else if (sortByPrice === "highToLow") {
+      return products.sort((a, b) => b.discountedPrice - a.discountedPrice);
+    }
+    return products;
+  };
+  const filterByRating = (products, rating) => {
+    console.log("rating", rating);
```
Comment (prathmeshjagtap) [inline src/Components/ProductList/ProductList.jsx]: remove console.log

File: src/Components/ProductList/ProductList.jsx [hunk 3]
```diff
@@ -1,14 +1,43 @@
...
+    return products;
+  };
+  const filteredProducts = filterByRating(
+    filterByCategory(
+      filterByPrice([...productList], state.sortByPrice),
+      state.category
+    ),
+    state.rating
+  );
+  console.log("filteredProducts", filteredProducts);
```
Comment (prathmeshjagtap) [inline src/Components/ProductList/ProductList.jsx]: remove console.log

File: src/Components/Sidebar/Sidebar.jsx [hunk 4]
```diff
@@ -17,45 +19,93 @@ const Sidebar = () => {
...
               </label>
               <label>
-                <input type="checkbox" name="category-filter" />
+                <input
+                  type="checkbox"
+                  name="category-filter"
+                  checked={state.category.includes("air-jordan")}
+                  onChange={() =>
+                    dispatch({ type: "CATEGORY", payload: "air-jordan" })
+                  }
```
Comment (prathmeshjagtap) [inline src/Components/Sidebar/Sidebar.jsx:67]: you can just create an array of categories  and map over it instead of repeating  code

**Review by tewarimohit — APPROVED** [review]
Good to merge..

**Review by prathmeshjagtap — CHANGES_REQUESTED** [review]

File: src/Components/Sidebar/Sidebar.jsx (hunk 4 above)
Comment (abhijeetekad) [reply src/Components/Sidebar/Sidebar.jsx:67]: Yes .. that's a great idea. will do that while refactoring.

File: src/Components/ProductList/ProductList.jsx (hunk 3 above)
Comment (abhijeetekad) [reply src/Components/ProductList/ProductList.jsx]: Yes..done

**Pull request merged**